File: UdemyMachineLearningAndAICourse/Polynomial Regression/polynomial_regression_CB.py
import numpy as mp 
import matplotlib.pyplot as plt 
import pandas as pd

#importing the dataSet
dataset =pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
Y = dataset.iloc[:,2].values


# Fit Linear Regression

from sklearn.linear_model import LinearRegression
linReg = LinearRegression()
linReg.fit(X,Y)

#Fit Polynomial Regression

from sklearn.preprocessing import PolynomialFeatures

poly_reg = PolynomialFeatures(degree=7)
X_poly = poly_reg.fit_transform(X)
linReg2 = LinearRegression()
linReg2.fit(X_poly,Y)

#Visuals for Linear 
plt.scatter(X,Y,color= 'green')
plt.plot(X,linReg.predict(X),color='red')
plt.title('Linear Presentation')
plt.show()
#Visuals for Polynomial 
X_grid = mp.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape((len(X_grid)),1)
plt.scatter(X,Y,color= 'green')
plt.plot(X_grid,linReg2.predict(poly_reg.fit_transform(X_grid)),color='blue')
plt.title('Polynomial Presentation')
plt.show()


#Predict result w/ Linear Reg
linReg.predict([[6.5]])
#Predict result w/ Polynomial Reg
linReg2.predict(poly_reg.fit_transform([[6.5]]))









#Split to training and test set
#from sklearn.model_selection import train_test_split
#X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size = 0.2,random_state = 0)

#Feature Scaling

""""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

File: UdemyMachineLearningAndAICourse/Polynomial Regression/polynomial_regression_CB.R
dataset = read.csv("Position_Salaries.csv")
#data needed from main
dataset = dataset[,2:3]

#split teaining and test set
#install.packages('caTools')
#set.seed(123)
#split= sample.split(dataset$Purchased,SplitRatio=0.8)
#training_set = subset(dataset,split==TRUE)
#test_set = subset(dataset,split==FALSE)

#FS
# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3] = scale(test_set[,2:3])

#fit Linear Regression to the dataset

lin_reg = lm(formula = Salary~ Level,data=dataset)
#fit Polynomial Regression to the dataset'
dataset$Level2 = dataset$Level ^2
dataset$Level3 = dataset$Level ^3
dataset$Level4 = dataset$Level ^4
#$Level5 = dataset$Level ^5

poly_reg = lm(formula = Salary~ .,data=dataset)




#Visuals for Linear
ggplot() + geom_point(aes (x=dataset$Level,y=dataset$Salary),colour= 'red')+geom_line( aes (x=dataset$Level,y= predict(lin_reg ,newdata = dataset),colour= 'blue'))  +ggtitle('Linear') +xlab('Level')

#Visuals for Polynomial
ggplot() + geom_point(aes (x=dataset$Level,y=dataset$Salary),colour= 'red')+geom_line( aes (x=dataset$Level,y= predict(poly_reg ,newdata = dataset),colour= 'blue'))  +ggtitle('Linear') +xlab('Level')



predict(lin_reg ,newdata = data.frame(Level = 6.5))
predict(poly_reg ,newdata =  data.frame(Level = 6.5,Level2 = 6.5^2,Level3= 6.5^3,Level4 = 6.5^4))
